Respect vs output map

JayFoxRox opened the pull request:
Something along these lines fixes most of #1504

What happens in Digimon?
* vs outmap_total is 5 = 5 output registers used
* vs outmap_mask is 0x2F (you might see where this is going..)
* shader wrote to o0, o1, o2, o3 and o5
* you know where the output semantics for o5 are configured? right.. vs outmap_o4
* said differently: outmap is packed tightly

Therefore I added a second index to count the actual outmap (output_attributes in citra) index.

As the file was already opened I switched the loop indices to unsigned.

I'm not sure about vs_output_total early out and it should be tested on actual HW.
I felt a little safer adding it because I wouldn't want possibly disabled (cause they are not included in output_total) regs overwriting outputs.

![Logo](http://i.imgur.com/XFjnCVS.png)
![Push my buttons](http://i.imgur.com/JX28ANM.png)
![Pushed me hard](http://i.imgur.com/qHaVtLK.png)
![Digimon stuff](http://i.imgur.com/AraiZQZ.png)
![Ash and pikachu](http://i.imgur.com/wna2Fjg.png)
![The dog still has broken eyes](http://i.imgur.com/1prdIqe.png)

I didn't know how to walk in the game as no button seemed to work so I didn't test if the 2D characters appear while talking.
There still seem to be some texturing issues but that's probably a seperate issue.

JohnGodgames commented:
### Also fixed:

Barbie And Her Sisters Puppy Rescue

![barbie puppy](https://cloud.githubusercontent.com/assets/11453507/13726640/dee577cc-e8d1-11e5-9f98-52c0472a2d02.JPG)

![barbie puppy 2](https://cloud.githubusercontent.com/assets/11453507/13726641/e7ac3a4e-e8d1-11e5-8d9e-47f464f90948.JPG)

Romsstar commented:
Confirming, this fixes almost all graphical issues of #1504, except for some textures like that Agumon above and also the 3D Model is not being shown on top screen during the save, it's supposed to.

Reference from 3DS again
![34d4bb62-e881-11e5-8be7-956c0aad237b](https://cloud.githubusercontent.com/assets/17484361/13726648/2ce17034-e8d2-11e5-8aca-e30f2d82beb9.png)

The 2D Images during talking also show. 
![fixefd](https://cloud.githubusercontent.com/assets/17484361/13726625/1bab0bdc-e8d1-11e5-9a7e-e3f1227610bf.png)

Instamerge please :D

MoochMcGee commented:
LGTM. Fixes stuff and it makes sense! Nice work!

JohnGodgames commented:
Also fixed solid color issued at these games:

Fireman Sam to The Rescue
Gardenscapes
Chibi Devi
Jewel Match 3 EUR
Penguins of Madagascar
Tiny Games - Knights & Dragons

LGTM

JayFoxRox commented:
Rebased to current master [a2024d749790f94979ad1f7a02363ab53f7de518]. Looks like #1496 also fixed Agumon texture in Digimon. Let's blame the menu rendering issue on depth issues or something which are not actually caused by this. (Also: The menu logo is rendered using sw rasterizer, which crashes on the save load screen - seperate issue)

![Fixed textures in master](http://i.imgur.com/kWqOnmj.png)

@citra-emu: We can merge this or check the outmap_total behaviour on hardware first (which is hard for me to do without direct access to hw). What do you prefer?

Romsstar commented:
Just tested this with #1496. It indeed fixes the agumon texture, but it now flickers sometimes.
I don't know though if it's my system or if that happens to others well.
Some 2D graphics still remain unfixed after this, but I believe this might be a separate issue.

Subv commented:
LGTM

JamePeng commented:
LGTM

bunnei commented:
LGTM

yuriks commented:
Did we run any hwtests on this?

JayFoxRox commented:
FTR: no.
